### ISSUE TITLE
Loadings change relative position to each other after rotation

### ISSUE THREAD

**Issue by gaardhus** [EducationalTestingService/factor_analyzer#77]
**Describe the bug**
I'm trying to orthogonally rotate loadings obtained by a PCA by using the rotator function, however the result changes the loadings change relative position to each other after rotation. This is more evident when more components are kept before rotation, and when only the two largest components are kept almost fits the original data.

**To Reproduce**
1. Download data https://we.tl/t-B3BJz8Gorz

```
from factor_analyzer import Rotator
import pandas as pd
import matplotlib.pyplot as plt
import numpy as np

# Read the loadings from the csv
df = pd.read_csv('loadings.csv', index_col=0)

# Plot the initial loadings
print('Original loadings')
df.plot(x=0, y=1, kind='scatter')
plt.axhline(0,color='black')
plt.axvline(0,color='black')
plt.xlim(-1,1)
plt.ylim(-1,1)
plt.show()

L = df.to_numpy()

# Loop over the number of components and remove the last each time

for _ in range(L.shape[1]-1):
    print(L.shape)

    rotator = Rotator(method='varimax', normalize=False)

    rotated_loadings = pd.DataFrame(rotator.fit_transform(L))

    rotated_loadings[[0,1]].plot(x=0, y=1, kind='scatter')
    plt.axhline(0,color='black')
    plt.axvline(0,color='black')
    plt.xlim(-1,1)
    plt.ylim(-1,1)
    plt.show()
    
    # Delete the smallest component
    L = np.delete(L, L.shape[1]-1, 1)
```

**Expected behavior**
I did not expect the loadings relative position to each other to change, neither for the number of components to have an impact on this. Maybe I have misunderstood varimax rotation, as I expected it to *only* rotate the axis, and not change the positions of the loadings in relation to each other, or I'm encountering some kind of bug.

**Comment by jbiggsets** [comment]
Thanks for the detailed explanation and for providing your loadings matrix and a code snippet.

I'm not sure I entirely understand the problem, but I'm pretty confident it isn't a bug in ``factor_analyzer``. I ran your loadings matrix through R's ``GPArotation`` library using a Varimax rotation and got exactly the same output as ``factor_analyzer``. (I tried this for only the first two components and all components.)

Just to make sure I understand, let's take your loop in **reverse** order. So, let's start with the last case, where you take only the two largest components from your original (unrotated) PCA, do a Varimax rotation, and the plot the output. 

When I look at the output here (which I reproduced), it's clear that the rotation is orthogonal, since the actual coordinate system hasn't changed.

<img width="799" alt="Screen Shot 2021-03-24 at 3 55 35 PM" src="https://user-images.githubusercontent.com/30508716/112375283-7e419e00-8cb9-11eb-8265-83ad3e0cb929.png">

I think the problem is that when you then start looking at three or more dimensions, you seem to only be **plotting** the first two. But if the rotation is happening in multiple dimensions, this is a misleading visualization. You'd need to visualize the _full_ coordinate system.

<img width="679" alt="Screen Shot 2021-03-24 at 4 01 22 PM" src="https://user-images.githubusercontent.com/30508716/112375899-3111fc00-8cba-11eb-93e4-baf262afa0f0.png">

This is still orthogonal, but the data are being rotated along three dimensions.

 Let me know if any of this doesn't make sense, or if I'm misunderstanding something. If it does make sense, l'll close this issue.
